### DISCUSSION TITLE
Overly cluttered add node menu

### DISCUSSION THREAD

**Discussion by Codesterz** [bnpr/Malt#160] [Feedback & Feature Requests]
I feel like the current node system could and should be de-cluttered. As an example in the vec3 and vec4 sub menus there are like 20 options. Instead of having so many choices broken up I think it would be better if they were done in a similar way to MixRGB in Eevee where you have a node with a drop down box and choose which option you want to do it from. As it is I have to scroll up and down to get to some of the options which is both time consuming and tedious. Now I don't think all the options in say vec4 should be put into a single node with a drop down box but try to reduce it to a couple nodes. Yes this will break current node tree's people have made, but it really should be done before the full release. The sooner the better.

EDIT: For nodes that would have a lot of options in the drop down menu maybe do something like the math node in cycles/eevee.

**Comment by pragma37** [comment]
I agree about the problem, but disagree with the solution.
You can see my answer to a similar request here:
https://github.com/bnpr/Malt/discussions/148#discussioncomment-1577052

What I typically do is simply use the search operator function to add new nodes.
So instead of browsing the vec3 menu for the add function, I type F3 > vec3 add > ENTER and I have my node.

Something I'd like to try is to support template-like nodes, so for example you have a single Add node that works for every data type that supports addition.
